plugins/org.eclipse.n4js/src/org/eclipse/n4js/generator/AbstractSubGenerator.xtend

```diff
@@ -196,6 +199,37 @@ abstract class AbstractSubGenerator implements ISubGenerator {
 		}
 	}
 
+	/** @return true iff the current project has a project type that is supposed to generate code. */
+	def boolean isGenerateProjectType(URI n4jsSourceURI) {
+		val project = n4jsCore.findProject(n4jsSourceURI).orNull();
+		if (project !== null) {
+			val projectType = project.getProjectType();
+			if (projectType == ProjectType.VALIDATION) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/** @return true iff the given resource does not lie within the output folder. */
+	def boolean isOutsideOfOutputFolder(URI n4jsSourceURI) {
```

Comment:
if it lies inside output folder and project type is not 'VALIDATION' then we silently skip compilation?
Maybe it would be better to add custom validation for this and not send those resources to the generator in the first place?